src/core/Header/Header.tsx

```diff
@@ -0,0 +1,206 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import '@itwin/itwinui-css/css/header.css';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import SvgHelpCircularHollow from '@itwin/itwinui-icons-react/cjs/icons/HelpCircularHollow';
+import SvgMoreVertical from '@itwin/itwinui-icons-react/cjs/icons/MoreVertical';
+import SvgNotification from '@itwin/itwinui-icons-react/cjs/icons/Notification';
+
+import cx from 'classnames';
+import React from 'react';
+import { IconButton } from '../Buttons';
+
+import { useTheme } from '../utils/hooks/useTheme';
+import { DropdownMenu } from '../DropdownMenu';
+
+export type HeaderTranslations = {
+  notifications: string;
+  help: string;
+  myAccount: string;
+  moreOptions: string;
+};
+
+export type HeaderProps = {
+  /**
+   * Icon to the application.
+   */
+  appIcon: React.ReactNode;
+  /**
+   * Name of the application.
+   */
+  appName?: React.ReactNode;
+  /**
+   * Handler on application click
+   */
+  onApplication?: () => void;
+  /**
+   * `HeaderButton` array, chevron will be added between each.
+   */
+  breadcrumbsButtons?: React.ReactNode[];
+  /**
+   * Content to display in the center section of the header, should handle iui-slim class if used.
+   */
+  centerContent?: React.ReactNode;
+  /**
+   * Expects `UserIcon`, will be given a drop down if `userMenuItems` is also provided;
+   */
+  userIcon?: React.ReactNode;
+  /**
+   * Handler on notification click, will show notification bell if provided.
+   */
+  onNotification?: () => void;
+  /**
+   * Items in the help dropdown menu.
+   * Pass a function that takes the `close` argument (to close the menu),
+   * and returns a list of `MenuItem` components.
+   */
+  helpMenuItems?: (close: () => void) => JSX.Element[];
```

Comment:
Depends on whether we want to provide more control or more simplicity

Reply:
Went with the function, simply because users will use them for the HeaderButton as they are now DropdownButtons, it seems cleaner to have all of these working in the same way. (However, we now only have 1 menuItems prop, for the "More options".)